### DISCUSSION TITLE
Web Gui runs but doesn't connect

### DISCUSSION THREAD

**Discussion by AliasJester** [haugene/docker-transmission-openvpn#2525] [Q&A]
Hi All, I've been searching and trying to work through this for the last 24 hours and hoping someone may be able to point out something I am missing.

I have Transmission working through NORD VPN in docker on my Synology. I can test it is working by placing a torrent in the watch directory - it downloads successfully. I have the web gui operating but when I open the web gui, the status just sits as "connecting..." and does not connect.

I'm in the 30000 port numbers just to test and have them forwarded on my router and in the docker port settings 30000:30000 and 30001:30001.

I am accessing the web gui at: http://192.168.0.118:30000/transmission/rpcweb/ and that gui comes up but does not connect to Transmission.

My settings.json and log are copied below, any help or pointers on where to look would be appreciated!

{
    "alt-speed-down": 50,
    "alt-speed-enabled": false,
    "alt-speed-time-begin": 540,
    "alt-speed-time-day": 127,
    "alt-speed-time-enabled": false,
    "alt-speed-time-end": 1020,
    "alt-speed-up": 50,
    "bind-address-ipv4": "10.7.1.6",
    "bind-address-ipv6": "::",
    "blocklist-enabled": false,
    "blocklist-url": "http://www.example.com/blocklist",
    "cache-size-mb": 4,
    "dht-enabled": true,
    "download-dir": "/data/completed",
    "download-queue-enabled": true,
    "download-queue-size": 5,
    "encryption": 1,
    "idle-seeding-limit": 30,
    "idle-seeding-limit-enabled": false,
    "incomplete-dir": "/data/incomplete",
    "incomplete-dir-enabled": true,
    "lpd-enabled": false,
    "message-level": 2,
    "peer-congestion-algorithm": "",
    "peer-id-ttl-hours": 6,
    "peer-limit-global": 240,
    "peer-limit-per-torrent": 60,
    "peer-port": 51413,
    "peer-port-random-high": 65535,
    "peer-port-random-low": 49152,
    "peer-port-random-on-start": false,
    "peer-socket-tos": "default",
    "pex-enabled": true,
    "port-forwarding-enabled": false,
    "preallocation": 1,
    "prefetch-enabled": true,
    "queue-stalled-enabled": true,
    "queue-stalled-minutes": 30,
    "ratio-limit": 2,
    "ratio-limit-enabled": false,
    "rename-partial-files": true,
    "rpc-authentication-required": false,
    "rpc-bind-address": "0.0.0.0",
    "rpc-enabled": true,
    "rpc-host-whitelist": "",
    "rpc-host-whitelist-enabled": false,
    "rpc-password": [redacted],
    "rpc-port": 30000,
    "rpc-url": "/transmission/rpc",
    "rpc-username": [redacted],
    "rpc-whitelist": "127.0.0.1,192.168.*.*,8.8.8.8",
    "rpc-whitelist-enabled": false,
    "scrape-paused-torrents-enabled": true,
    "script-torrent-done-enabled": false,
    "script-torrent-done-filename": "",
    "seed-queue-enabled": false,
    "seed-queue-size": 10,
    "speed-limit-down": 100,
    "speed-limit-down-enabled": false,
    "speed-limit-up": 100,
    "speed-limit-up-enabled": false,
    "start-added-torrents": true,
    "trash-original-torrent-files": false,
    "umask": 2,
    "upload-slots-per-torrent": 14,
    "utp-enabled": false,
    "watch-dir": "/data/watch",
    "watch-dir-enabled": true,
    "watch-dir-force-generic": false
}

2023-02-08 01:46:08 net_route_v4_add: 128.0.0.0/1 via 10.7.3.1 dev [NULL] table 0 metric -1

Up script executed with device=tun0 ifconfig_local=10.7.3.2

Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 10.7.3.2

Using Transmission Web Control UI, overriding TRANSMISSION_WEB_HOME

TRANSMISSION_HOME is currently set to: /config/transmission-home

WARNING: Deprecated. Found old default transmission-home folder at /data/transmission-home, setting this as TRANSMISSION_HOME. This might break in future versions.

We will fallback to this directory as long as the folder exists. Please consider moving it to /config/<transmission-home>

Enforcing ownership on transmission config directory

Applying permissions to transmission config directory

Setting owner for transmission paths to 1003:1003

Setting permissions for download and incomplete directories

Mask: 002

Directories: 775

Files: 664

Setting permission for watch directory (775) and its files (664)

-------------------------------------

Transmission will run as

-------------------------------------

User name:   abc

User uid:    1003

User gid:    1003

-------------------------------------

Updating Transmission settings.json with values from env variables

Attempting to use existing settings.json for Transmission

Successfully used existing settings.json /data/transmission-home/settings.json

Overriding bind-address-ipv4 because TRANSMISSION_BIND_ADDRESS_IPV4 is set to 10.7.3.2

Overriding download-dir because TRANSMISSION_DOWNLOAD_DIR is set to /data/completed

Overriding incomplete-dir because TRANSMISSION_INCOMPLETE_DIR is set to /data/incomplete

Overriding rpc-authentication-required because TRANSMISSION_RPC_AUTHENTICATION_REQUIRED is set to false

Overriding rpc-enabled because TRANSMISSION_RPC_ENABLED is set to true

Overriding rpc-password because TRANSMISSION_RPC_PASSWORD is set to [REDACTED]

Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 30000

Overriding rpc-url because TRANSMISSION_RPC_URL is set to /transmission/rpc

Overriding rpc-username because TRANSMISSION_RPC_USERNAME is set to [redacted]

Overriding rpc-whitelist because TRANSMISSION_RPC_WHITELIST is set to 127.0.0.1,192.168.*.*,8.8.8.8

Overriding rpc-whitelist-enabled because TRANSMISSION_RPC_WHITELIST_ENABLED is set to false

Overriding umask because TRANSMISSION_UMASK is set to 2

Overriding watch-dir because TRANSMISSION_WATCH_DIR is set to /data/watch

sed'ing True to true

DROPPING DEFAULT ROUTE

STARTING TRANSMISSION

Transmission startup script complete.

Privoxy: Starting

Privoxy: Using config file at /etc/privoxy/config

Privoxy: Setting port to 30001

[2023-02-08 01:46:09.799] Transmission 3.00 (bb6b5a062e) started (session.c:769)

[2023-02-08 01:46:09.799] RPC Server Adding address to whitelist: 127.0.0.1 (rpc-server.c:956)

[2023-02-08 01:46:09.799] RPC Server Adding address to whitelist: 192.168.*.* (rpc-server.c:956)

[2023-02-08 01:46:09.799] RPC Server Adding address to whitelist: 8.8.8.8 (rpc-server.c:956)

[2023-02-08 01:46:09.799] RPC Server Serving RPC and Web requests on 0.0.0.0:30000/transmission/rpc (rpc-server.c:1243)

[2023-02-08 01:46:09.799] Port Forwarding Stopped (port-forwarding.c:196)

[2023-02-08 01:46:09.799] DHT Reusing old id (tr-dht.c:383)

[2023-02-08 01:46:09.799] DHT Bootstrapping from 116 IPv4 nodes (tr-dht.c:172)

[2023-02-08 01:46:09.799] Using settings from "/data/transmission-home" (daemon.c:646)

[2023-02-08 01:46:09.799] Saved "/data/transmission-home/settings.json" (variant.c:1221)

[2023-02-08 01:46:09.799] Watching "/data/watch" for new .torrent files (daemon.c:698)

[2023-02-08 01:46:09.799] Loaded 1 torrents (session.c:2170)

[2023-02-08 01:46:09.799] web will verify tracker certs using envvar CURL_CA_BUNDLE: none (web.c:455)

[2023-02-08 01:46:09.799] web NB: this only works if you built against libcurl with openssl or gnutls, NOT nss (web.c:457)

[2023-02-08 01:46:09.799] web NB: invalid certs will show up as 'Could not connect to tracker' like many other errors (web.c:458)

Privoxy: Running as PID 147

2023-02-08 01:46:10 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this

2023-02-08 01:46:10 Initialization Sequence Completed

**Comment by pkishino** [comment]
you don't need any port forwarding on your router..
what is port 30001 used for?
default access to the ui should be <ip-of-container>:<port>
you can try adding /transmission/web/ at the end..

**Reply by AliasJester** [reply]
Thanks. 

WEBPROXY_PORT = 30000
WEBPROXY_ENABLE = true
These are set in environment variables but I just noticed they are not captured in my settings.json

192.168.0.118:30000/transmission/web/ redirects me to 192.168.0.118:30000/transmission/rpcweb/ and that doesn't connect to Transmission

If I use http://192.168.0.118:30000/transmission/rpc/ I get a 409 Conflict

Do you know if this 409 Conflict is what I should be trying to resolve?

```
409: Conflict

Your request had an invalid session-id header.

To fix this, follow these steps:

    When reading a response, get its X-Transmission-Session-Id header and remember it
    Add the updated header to your outgoing requests
    When you get this 409 error message, resend your request with the updated header

This requirement has been added to help prevent [CSRF](https://en.wikipedia.org/wiki/Cross-site_request_forgery) attacks.
```

**Reply by pkishino** [reply]
settings.json is just for transmission settings.. the web proxy has nothing to do with that.
please show your whole docker config..

**Reply by AliasJester** [reply]
Thanks. Docker settings pasted below.

{
   "CapAdd" : [],
   "CapDrop" : [],
   "cmd" : "dumb-init /etc/openvpn/start.sh",
   "cpu_priority" : 50,
   "enable_publish_all_ports" : false,
   "enable_restart_policy" : false,
   "enable_service_portal" : null,
   "enabled" : false,
   "env_variables" : [
      {
         "key" : "PATH",
         "value" : "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      },
      {
         "key" : "OPENVPN_USERNAME",
         "value" : "REDACTED"
      },
      {
         "key" : "OPENVPN_PASSWORD",
         "value" : "REDACTED"
      },
      {
         "key" : "OPENVPN_PROVIDER",
         "value" : "NORDVPN"
      },
      {
         "key" : "GLOBAL_APPLY_PERMISSIONS",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_HOME",
         "value" : "/config/transmission-home"
      },
      {
         "key" : "TRANSMISSION_RPC_PORT",
         "value" : "30000"
      },
      {
         "key" : "TRANSMISSION_RPC_USERNAME",
         "value" : "REDACTED"
      },
      {
         "key" : "TRANSMISSION_RPC_PASSWORD",
         "value" : "REDACTED"
      },
      {
         "key" : "TRANSMISSION_DOWNLOAD_DIR",
         "value" : "/data/completed"
      },
      {
         "key" : "TRANSMISSION_INCOMPLETE_DIR",
         "value" : "/data/incomplete"
      },
      {
         "key" : "TRANSMISSION_WATCH_DIR",
         "value" : "/data/watch"
      },
      {
         "key" : "CREATE_TUN_DEVICE",
         "value" : "true"
      },
      {
         "key" : "ENABLE_UFW",
         "value" : "false"
      },
      {
         "key" : "UFW_ALLOW_GW_NET",
         "value" : "false"
      },
      {
         "key" : "UFW_EXTRA_PORTS",
         "value" : "30000,30001"
      },
      {
         "key" : "UFW_DISABLE_IPTABLES_REJECT",
         "value" : "false"
      },
      {
         "key" : "PUID",
         "value" : "1003"
      },
      {
         "key" : "PGID",
         "value" : "1003"
      },
      {
         "key" : "PEER_DNS",
         "value" : "true"
      },
      {
         "key" : "PEER_DNS_PIN_ROUTES",
         "value" : "true"
      },
      {
         "key" : "DROP_DEFAULT_ROUTE",
         "value" : "true"
      },
      {
         "key" : "WEBPROXY_ENABLED",
         "value" : "true"
      },
      {
         "key" : "WEBPROXY_PORT",
         "value" : "30001"
      },
      {
         "key" : "WEBPROXY_USERNAME",
         "value" : "REDACTED"
      },
      {
         "key" : "WEBPROXY_PASSWORD",
         "value" : "REDACTED"
      },
      {
         "key" : "LOG_TO_STDOUT",
         "value" : "true"
      },
      {
         "key" : "HEALTH_CHECK_HOST",
         "value" : "google.com"
      },
      {
         "key" : "SELFHEAL",
         "value" : "false"
      },
      {
         "key" : "REVISION",
         "value" : "REDACTED"
      },
      {
         "key" : "OPENVPN_OPTS",
         "value" : "--inactive 3600 --ping 10 --ping-exit 60 --dhcp-option DNS 8.8.8.8"
      },
      {
         "key" : "TRANSMISSION_RPC_URL",
         "value" : "/transmission/rpc"
      },
      {
         "key" : "NORDVPN_PROTOCOL",
         "value" : "tcp"
      },
      {
         "key" : "NORDVPN_CATEGORY",
         "value" : "legacy_p2p"
      },
      {
         "key" : "NORDVPN_COUNTRY",
         "value" : "au"
      },
      {
         "key" : "TRANSMISSION_RPC_ENABLED",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_RPC_AUTHENTICATION_REQUIRED",
         "value" : "false"
      },
      {
         "key" : "TRANSMISSION_RPC_WHITELIST_ENABLED",
         "value" : "false"
      },
      {
         "key" : "LOCAL_NETWORK",
         "value" : "192.168.0.0/24"
      },
      {
         "key" : "TRANSMISSION_WEB_UI",
         "value" : "transmission-web-control"
      },
      {
         "key" : "TRANSMISSION_RPC_WHITELIST",
         "value" : "127.0.0.1,192.168.*.*,8.8.8.8"
      },
      {
         "key" : "TRANSMISSION_UMASK",
         "value" : "2"
      },
      {
         "key" : "DISABLE_PORT_UPDATER",
         "value" : "true"
      }
   ],
   "exporting" : false,
   "id" : "17a936ec6edf348e7be10723960e36a5610f399d56a41bbc9761fbe6e1725179",
   "image" : "haugene/transmission-openvpn:edge",
   "is_ddsm" : false,
   "is_package" : false,
   "links" : [],
   "memory_limit" : 0,
   "name" : "haugene-transmission-openvpn",
   "network" : [
      {
         "driver" : "bridge",
         "name" : "bridge"
      }
   ],
   "network_mode" : "bridge",
   "port_bindings" : [
      {
         "container_port" : 30000,
         "host_port" : 30000,
         "type" : "tcp"
      },
      {
         "container_port" : 30001,
         "host_port" : 30001,
         "type" : "tcp"
      }
   ],
   "privileged" : true,
   "shortcut" : {
      "enable_shortcut" : false,
      "enable_status_page" : false,
      "enable_web_page" : false,
      "web_page_url" : ""
   },
   "use_host_network" : false,
   "volume_bindings" : [
      {
         "host_volume_file" : "/Storage/Downloads",
         "mount_point" : "/data",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/docker/transmission-openvpn/resolve.conf",
         "mount_point" : "/etc/resolve.conf",
         "type" : "rw"
      }
   ]
}

**Comment by pkishino** [comment]
1)
fix below 
```
{
"key" : "TRANSMISSION_RPC_WHITELIST",
"value" : "127.0.0.1,192.168..,8.8.8.8"
},
```
second value is incorrect

2) check your /data/ folder and remove transmission config folder from there (you already mapped a /config volume..)
3) try after above, else, stop container, delete your transmission settings.json file, then restart

access should be fine via http://<container-ip>:3000
no path added

**Comment by AliasJester** [comment]
Thank for this.

1. Asterisks aren't showing in these git comments for some reason but they are there in the file. The address is 192.168.asterisk.asterisk.

2. I have deleted the "transmission-home" folder from /data/. If I remake the empty "transmission-home" folder in the mapped location, it does get populated again. This is one part where my understanding of docker and this system breaks down. Where actually is "/config/transmission-home"?

This didn't fix the issue though unfortunately.

3. I have deleted settings.json a number of times and again now. Still not working.

I have just tried a new docker install and I seem to have it working. Thanks so much for answering my questions. I'm going to test this new install and then I'll report back on the settings that I have used to make this work. I think I may have added too many non-default environment variables in my last install and caused a conflict or error somewhere.

**Reply by AliasJester** [reply]
Appreciate the support pkishino!

For anyone coming across the same issues I have pasted my new settings below. It's probably pretty standard and I must have messed something up originally.

Beyond the stock settings for anyone looking at this I added:
- LOCAL_NETWORK
- TRANSMISSION_RPC_ENABLED

Earlier I also had an issue where I was getting an "RPC Server Unable to bind" error. That was my mistake with incorrectly setting the PUID and PGID values. Just noting for reference.

```
{
   "CapAdd" : [],
   "CapDrop" : [],
   "cmd" : "dumb-init /etc/openvpn/start.sh",
   "cpu_priority" : 50,
   "enable_publish_all_ports" : false,
   "enable_restart_policy" : false,
   "enable_service_portal" : false,
   "enabled" : true,
   "env_variables" : [
      {
         "key" : "PATH",
         "value" : "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
      },
      {
         "key" : "OPENVPN_USERNAME",
         "value" : "REDACTED"
      },
      {
         "key" : "OPENVPN_PASSWORD",
         "value" : "REDACTED"
      },
      {
         "key" : "OPENVPN_PROVIDER",
         "value" : "NORDVPN"
      },
      {
         "key" : "OPENVPN_OPTS",
         "value" : "--inactive 3600 --ping 10 --ping-exit 60 --dhcp-option DNS 8.8.8.8"
      },
      {
         "key" : "GLOBAL_APPLY_PERMISSIONS",
         "value" : "true"
      },
      {
         "key" : "TRANSMISSION_HOME",
         "value" : "/config/transmission-home"
      },
      {
         "key" : "TRANSMISSION_RPC_PORT",
         "value" : "9091"
      },
      {
         "key" : "TRANSMISSION_RPC_USERNAME",
         "value" : "transmission"
      },
      {
         "key" : "TRANSMISSION_RPC_PASSWORD",
         "value" : "transmission"
      },
      {
         "key" : "TRANSMISSION_DOWNLOAD_DIR",
         "value" : "/data/completed"
      },
      {
         "key" : "TRANSMISSION_INCOMPLETE_DIR",
         "value" : "/data/incomplete"
      },
      {
         "key" : "TRANSMISSION_WATCH_DIR",
         "value" : "/data/watch"
      },
      {
         "key" : "CREATE_TUN_DEVICE",
         "value" : "true"
      },
      {
         "key" : "ENABLE_UFW",
         "value" : "false"
      },
      {
         "key" : "UFW_ALLOW_GW_NET",
         "value" : "false"
      },
      {
         "key" : "UFW_EXTRA_PORTS",
         "value" : "8118,9091"
      },
      {
         "key" : "UFW_DISABLE_IPTABLES_REJECT",
         "value" : "false"
      },
      {
         "key" : "PUID",
         "value" : "1003"
      },
      {
         "key" : "PGID",
         "value" : "1003"
      },
      {
         "key" : "PEER_DNS",
         "value" : "true"
      },
      {
         "key" : "PEER_DNS_PIN_ROUTES",
         "value" : "true"
      },
      {
         "key" : "DROP_DEFAULT_ROUTE",
         "value" : "false"
      },
      {
         "key" : "WEBPROXY_ENABLED",
         "value" : "true"
      },
      {
         "key" : "WEBPROXY_PORT",
         "value" : "8118"
      },
      {
         "key" : "WEBPROXY_USERNAME",
         "value" : ""
      },
      {
         "key" : "WEBPROXY_PASSWORD",
         "value" : ""
      },
      {
         "key" : "LOG_TO_STDOUT",
         "value" : "false"
      },
      {
         "key" : "HEALTH_CHECK_HOST",
         "value" : "google.com"
      },
      {
         "key" : "SELFHEAL",
         "value" : "false"
      },
      {
         "key" : "REVISION",
         "value" : "b33d0fe4c938259a0d4eb844e55468f387456121"
      },
      {
         "key" : "LOCAL_NETWORK",
         "value" : "192.168.0.0/24"
      },
      {
         "key" : "TRANSMISSION_RPC_ENABLED",
         "value" : "true"
      }
   ],
   "exporting" : false,
   "id" : "8e2d63d8e919d3e6bef785a47259d20efcdd22dee08ed32f8a95e0ff54f7ba20",
   "image" : "haugene/transmission-openvpn:latest",
   "is_ddsm" : false,
   "is_package" : false,
   "links" : [],
   "memory_limit" : 0,
   "name" : "haugene-transmission-openvpn2",
   "network" : [
      {
         "driver" : "bridge",
         "name" : "bridge"
      }
   ],
   "network_mode" : "bridge",
   "port_bindings" : [
      {
         "container_port" : 8118,
         "host_port" : 8118,
         "type" : "tcp"
      },
      {
         "container_port" : 9091,
         "host_port" : 9091,
         "type" : "tcp"
      }
   ],
   "privileged" : true,
   "service_portals" : [],
   "shortcut" : {
      "enable_shortcut" : false,
      "enable_status_page" : false,
      "enable_web_page" : false,
      "web_page_url" : ""
   },
   "use_host_network" : false,
   "volume_bindings" : [
      {
         "host_volume_file" : "/Storage/Downloads",
         "mount_point" : "/data/",
         "type" : "rw"
      },
      {
         "host_volume_file" : "/docker/transmission-openvpn/resolve.conf",
         "mount_point" : "/docker/transmission-openvpn/resolve.conf",
         "type" : "rw"
      }
   ]
}
```
